SmartDeviceLink_Example/MenuManager.swift

```diff
@@ -0,0 +1,193 @@
+            manager.send(AlertManager.alertWithMessageAndCloseButton("Speech recognition feature only available on iOS 10+"))
+        })
+    }
+
+    /// Menu item that dials a phone number when selected
+    ///
+    /// - Parameter manager: The SDL Manager
+    /// - Returns: A SDLMenuCell object
+    class func menuCellDialNumber(with manager: SDLManager) -> SDLMenuCell {
+        return SDLMenuCell(title: ACDialPhoneNumberMenuName, icon: SDLArtwork(image: UIImage(named: PhoneBWIconImageName)!, persistent: true, as: .PNG), voiceCommands: [ACDialPhoneNumberMenuName], handler: { (triggerSource) in
+            guard RPCPermissionsManager.isDialNumberRPCAllowed(with: manager) else {
+                manager.send(AlertManager.alertWithMessageAndCloseButton("This app does not have the required permissions to dial a number"))
+                return
+            }
+
+            VehicleDataManager.checkPhoneCallCapability(manager: manager, phoneNumber:"555-555-5555")
+        })
+    }
+
+    /// Menu item that opens a submenu when selected
+    ///
+    /// - Parameter manager: The SDL Manager
+    /// - Returns: A SDLMenuCell object
+    class func menuCellWithSubmenu(with manager: SDLManager) -> SDLMenuCell {
+        var submenuItems = [SDLMenuCell]()
+        for i in 0..<75 {
+            let submenuTitle = "Submenu Item \(i)"
+            submenuItems.append(SDLMenuCell(title: submenuTitle, icon: SDLArtwork(image: UIImage(named: MenuBWIconImageName)!, persistent: true, as: .PNG), voiceCommands: [submenuTitle, "Item \(i)", "\(i)"], handler: { (triggerSource) in
+                manager.send(AlertManager.alertWithMessageAndCloseButton("Submenu item \(i) selected!"))
+            }))
+        }
+
+        return SDLMenuCell(title: "Submenu", subCells: submenuItems)
+    }
+}
+
+// MARK: - Perform Interaction Choice Set Menu
+
+private extension MenuManager {
+    static let choiceSetId = 100
+    /// The PICS menu items
+    ///
+    /// - Returns: An array of SDLChoice items
+    class func createChoiceSet() -> [SDLChoice] {
+        let firstChoice = SDLChoice(id: 1, menuName: PICSFirstChoice, vrCommands: [PICSFirstChoice])
+        let secondChoice = SDLChoice(id: 2, menuName: PICSSecondChoice, vrCommands: [PICSSecondChoice])
+        let thirdChoice = SDLChoice(id: 3, menuName: PICSThirdChoice, vrCommands: [PICSThirdChoice])
+        return [firstChoice, secondChoice, thirdChoice]
+    }
+
+    /// Creates a PICS with three menu items and customized voice commands
+    ///
+    /// - Returns: A SDLPerformInteraction request
+    class func createPerformInteraction() -> SDLPerformInteraction {
+        let performInteraction = SDLPerformInteraction(initialPrompt: PICSInitialPrompt, initialText: PICSInitialText, interactionChoiceSetIDList: [choiceSetId as NSNumber], helpPrompt: PICSHelpPrompt, timeoutPrompt: PICSTimeoutPrompt, interactionMode: .both, timeout: 10000)
```

Comment:
The `interactionMode` should come from the `triggerSource` of the menu selection. If touched -> Manual only, voice -> voice only.

Reply:
Fixed in both Swift and Obj-C Ex. apps.